treasury-fees-monitor/data-collection/collect-fee-data.js

```diff
@@ -0,0 +1,183 @@
+const commandLineArgs = require('command-line-args');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const ethers = require('ethers');
+const fs = require('fs');
+
+require('dotenv').config();
+
+const BLOCK_CHUNK = 128;
+const SECONDS_IN_DAY = 24 * 60 * 60;
+
+async function getBlockByTimestamp(provider, userTimestamp) {
+  // seed the algorithm
+  // starting value for low block
+  let lowBlockNumber = 1;
+  let lowBlock = await provider.getBlock(lowBlockNumber);
+  let lowBlockTimestamp = lowBlock.timestamp;
+
+  // starting value for high block
+  let highBlock = await provider.getBlock('latest');
+  let highBlockNumber = highBlock.number;
+  let highBlockTimestamp = highBlock.timestamp;
+
+  // iterate until we find the two blocks whose timestamps bound the requested timestamp
+  let blockPerTime = 0;
+  let newBlock;
+  let newBlockNumber;
+  let newBlockTimestamp;
+  while (highBlockNumber - lowBlockNumber > 1 && lowBlockTimestamp !== newBlockTimestamp) {
+    // calculate average number of blocks per second
+    blockPerTime = (highBlockNumber - lowBlockNumber) / (highBlockTimestamp - lowBlockTimestamp);
+
+    // estimate the correct block number where the user's timestamp should exist
+    newBlockNumber = Math.floor(
+      lowBlockNumber + (blockPerTime * (userTimestamp - lowBlockTimestamp)),
+    );
+
+    // retrieve the block and timestamp
+    // eslint-disable-next-line no-await-in-loop
+    newBlock = await provider.getBlock(newBlockNumber);
+    newBlockTimestamp = newBlock.timestamp;
+
+    if (newBlockTimestamp > userTimestamp) {
+      // if the retrieved timestamp is higher than we expected, shift the high block down
+      highBlock = newBlock;
+      highBlockTimestamp = newBlockTimestamp;
+      highBlockNumber = newBlockNumber;
+    } else if (newBlockTimestamp < userTimestamp) {
+      // if the retrieved timestamp is lower than we expected, shift the high block down
```

Comment:
```suggestion
      // if the retrieved timestamp is lower than we expected, shift the low block down
```

Did you mean to say shift the low block instead of high?